test-tools/wamr-ide/VSCode-Extension/src/utilities/dockerUtilities.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2019 Intel Corporation.  All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+import * as vscode from 'vscode';
+import * as cp from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs';
+import { getWAMRExtensionVersion } from './lldbUtilities';
+import { downloadFile, unzipFile } from './directoryUtilities';
+
+const DOCKER_IMAGES_TEM_FOLDER_NAME = 'docker-resource';
+
+const execShell = (cmd: string) =>
+    new Promise<string>((resolve, reject) => {
+        cp.exec(cmd, (error, result) => {
+            if (error) {
+                return reject(error);
+            }
+            return resolve(result);
+        });
+    });
+
+export async function promptSetupDockerImages(
+    context: vscode.ExtensionContext
+): Promise<string> {
+    const extensionPath = context.extensionPath;
+    const setupPrompt = 'setup';
+    const skipPrompt = 'skip';
+    const response = await vscode.window.showWarningMessage(
+        'Necessary docker images are not found. Setup now?',
+        setupPrompt,
+        skipPrompt
+    );
+
+    if (response === skipPrompt) {
+        return Promise.resolve(skipPrompt);
+    }
+
+    const downloadUrlArray = getDockerImagesDownloadUrl(context);
+
+    const destinationFolder = path.resolve(
+        extensionPath,
+        'resource',
+        DOCKER_IMAGES_TEM_FOLDER_NAME
+    );
+
+    if (!fs.existsSync(destinationFolder)) {
```

Comment:
That technically only checks if file or directory exists, whereas we want to make sure this is a directory. I don't think it's a big problem though, so we can probably leave it as it is for now.